nodefs/loopback_linux_test.go

```diff
@@ -198,3 +202,119 @@ func TestCopyFileRange(t *testing.T) {
 	}
 
 }
+
+// Wait for a change in /proc/self/mounts
+func waitMount() error {
+	fd, err := syscall.Open("/proc/self/mounts", syscall.O_RDONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer syscall.Close(fd)
+	pollFds := []unix.PollFd{
+		{
+			Fd:     int32(fd),
+			Events: unix.POLLPRI,
+		},
+	}
+	_, err = unix.Poll(pollFds, 1000)
+	return err
+}
+
+// There is a hang that appears when enabling CAP_PARALLEL_DIROPS on Linux
+// 4.15.0: https://github.com/hanwen/go-fuse/issues/281
+// The hang was originally triggered by gvfs-udisks2-volume-monitor. This
+// test emulates what gvfs-udisks2-volume-monitor does.
+func TestParallelDiropsHang(t *testing.T) {
+	// We do NOT want to use newTestCase() here because we need access to
+	// mntDir before the filesystem is mounted
```

Comment:
@rfjakob, could you please clarify where/how mnt directory is accessed before the filesystem is mounted? (I'm looking at the test and see access to mnt/* only after waitMount completes)

Reply:
Reworded - we need to know the mountpoint path before it is mounted. We don't access the directory there, we just need to know the path.